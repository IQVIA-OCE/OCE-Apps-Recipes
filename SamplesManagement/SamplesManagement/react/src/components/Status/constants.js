export const statusColors = {
  InProgress: {
    color: '#ffb75d',
  },
  Urgent: {
    color: '#ff6565',
  },
  Submitted: {
    color: '#04844b',
  },
  Approved: {
    color: '#04844b',
  },
  Saved: {
    color: '#5eb4ff',
  },
  Rejected: {
    color: '#c23934',
  },
  Shipped: {
    color: '#04844b',
  },
  BackOrder: {
    color: '#5eb4ff',
  },
  SentToFulfillment: {
    color: '#04844b',
  },
  PendingApproval: {
    color: '#5eb4ff',
  },
  SubmittedforApproval: {
    color: '#5eb4ff',
  },
  Returned: {
    color: '#c23934',
  },
  Duplicate: {
    color: '#5eb4ff',
  },
  Info: {
    color: '#ecebea',
    textColor: '#000'
  }
};
