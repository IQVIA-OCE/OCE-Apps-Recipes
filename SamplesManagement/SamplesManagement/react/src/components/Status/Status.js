import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { statusColors } from './constants';

const getStatusLabelStyles = status => {
  return status ? statusColors[status.replace(' ', '')] : '';
};

const Status = ({ status, size, text }) => {
  return (
    <View
      style={[
        styles.container,
        styles[size],
        {
          backgroundColor: getStatusLabelStyles(status).color,
          marginRight: 5,
        },
      ]}
    >
      <Text
        style={[
          styles.labelText,
          {
            color: getStatusLabelStyles(status).textColor
              ? getStatusLabelStyles(status).textColor
              : 'white',
          },
        ]}
      >
        {text ? text : status}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 2,
    paddingHorizontal: 6,
    borderRadius: 10,
    alignSelf: 'flex-start',
  },
  labelText: {
    fontSize: 12,
    fontWeight: '700',
    color: 'white',
  },
  large: {
    fontSize: 16,
    paddingVertical: 6,
    paddingHorizontal: 8,
    borderRadius: 15,
  },
});

export default Status;
