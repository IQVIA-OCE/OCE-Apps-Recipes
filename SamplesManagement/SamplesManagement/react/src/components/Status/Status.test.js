import React from 'react';
import renderer from 'react-test-renderer';
import Status from './Status';

describe('Status', () => {
  it('Should render Status InProgress', () => {
    const tree = renderer
      .create(
        <Status status={'InProgress'}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
