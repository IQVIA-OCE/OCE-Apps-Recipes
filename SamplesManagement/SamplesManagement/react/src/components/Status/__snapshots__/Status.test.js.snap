// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Should render Status InProgress 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderRadius": 10,
        "flexDirection": "row",
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      undefined,
      Object {
        "backgroundColor": "#ffb75d",
        "marginRight": 5,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "700",
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    InProgress
  </Text>
</View>
`;
